modules/UI/videolayout/LocalVideo.js

```diff
@@ -112,8 +112,11 @@ LocalVideo.prototype.changeVideo = function (stream) {
         localVideoContainer.removeChild(localVideo);
         // when removing only the video element and we are on stage
         // update the stage
-        if(this.isCurrentlyOnLargeVideo())
-            this.VideoLayout.updateLargeVideo(this.id);
+        if (this.isCurrentlyOnLargeVideo()) {
+            this.VideoLayout.updateLargeVideo(
+                this.id,
+                true /* force - stream removed for the same user ID */);
+        }
```

Comment:
No, the condition is to update if the id is *not* on large video or 'force'. The fix is for the case when user had no video and the video is being added. Previously the update worked by itself because the ID was null when user was without video stream, but now the id is not obtained through the stream and is always defined.

I will try to look into moving the 'force update' decision for such case to inside of VideoLayout.